File: android/src/main/java/com/bitmovin/player/reactnative/DebugExpoModule.kt
package com.bitmovin.player.reactnative

import com.bitmovin.player.api.DebugConfig
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class DebugExpoModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("DebugModule")

        AsyncFunction("setDebugLoggingEnabled") { enabled: Boolean ->
            DebugConfig.isLoggingEnabled = enabled
        }
    }
}

File: ios/DebugExpoModule.swift
import BitmovinPlayer
import ExpoModulesCore

public class DebugExpoModule: Module {
    public func definition() -> ModuleDefinition {
        Name("DebugModule")

        AsyncFunction("setDebugLoggingEnabled") { (enabled: Bool) in
            DebugConfig.logging.logger?.level = enabled ? .verbose : .warning
        }.runOnQueue(.main)
    }
}
